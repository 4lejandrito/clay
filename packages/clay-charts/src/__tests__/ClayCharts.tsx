/**
 * Â© 2019 Liferay, Inc. <https://liferay.com>
 *
 * SPDX-License-Identifier: BSD-3-Clause
 */

import ClayCharts from '..';
import React from 'react';
import {cleanup, render} from 'react-testing-library';

const COLUMNS = [['data1', 100, 20, 30], ['data2', 20, 70, 100]];

describe('ClayCharts', () => {
	afterEach(cleanup);

	it('renders', () => {
		const {container} = render(
			<ClayCharts
				data={{
					columns: COLUMNS,
				}}
			/>
		);

		expect(container).toMatchSnapshot();
	});
});
