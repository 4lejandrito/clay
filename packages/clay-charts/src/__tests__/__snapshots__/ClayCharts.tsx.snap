// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClayCharts renders 1`] = `
<div>
  <div
    class="bb"
    style="position: relative;"
  >
    <svg
      height="320"
      style="overflow: hidden; display: block;"
      width="0"
    >
      <defs>
        <clippath
          id="bb-1560811908868-clip"
        >
          <rect
            height="226"
            width="0"
          />
        </clippath>
        <clippath
          id="bb-1560811908868-clip-xaxis"
        >
          <rect
            height="110"
            width="62"
            x="-31"
            y="-20"
          />
        </clippath>
        <clippath
          id="bb-1560811908868-clip-yaxis"
        >
          <rect
            height="250"
            width="40"
            x="-29"
            y="-4"
          />
        </clippath>
        <clippath
          id="bb-1560811908868-clip-grid"
        >
          <rect
            height="226"
            width="0"
          />
        </clippath>
      </defs>
      <g
        transform="translate(20.5, 4.5)"
      >
        <rect
          class="bb-zoom-rect"
          height="226"
          style="opacity: 0;"
          width="0"
        />
        <g
          class="bb-regions"
          clip-path="url(http://localhost/#bb-1560811908868-clip)"
          style="visibility: visible;"
        />
        <g
          class="bb-grid"
          clip-path="url(http://localhost/#bb-1560811908868-clip-grid)"
          style="visibility: visible;"
        >
          <g
            class="bb-xgrids"
          >
            <line
              class="bb-xgrid"
              style="opacity: 1;"
              x1="1"
              x2="1"
              y1="0"
              y2="226"
            />
            <line
              class="bb-xgrid"
              style="opacity: 1;"
              x1="1"
              x2="1"
              y1="0"
              y2="226"
            />
            <line
              class="bb-xgrid"
              style="opacity: 1;"
              x1="1"
              x2="1"
              y1="0"
              y2="226"
            />
          </g>
          <g
            class="bb-ygrids"
          >
            <line
              class="bb-ygrid"
              x1="0"
              x2="0"
              y1="207.25"
              y2="207.25"
            />
            <line
              class="bb-ygrid"
              x1="0"
              x2="0"
              y1="183.8125"
              y2="183.8125"
            />
            <line
              class="bb-ygrid"
              x1="0"
              x2="0"
              y1="160.375"
              y2="160.375"
            />
            <line
              class="bb-ygrid"
              x1="0"
              x2="0"
              y1="136.9375"
              y2="136.9375"
            />
            <line
              class="bb-ygrid"
              x1="0"
              x2="0"
              y1="113.5"
              y2="113.5"
            />
            <line
              class="bb-ygrid"
              x1="0"
              x2="0"
              y1="90.0625"
              y2="90.0625"
            />
            <line
              class="bb-ygrid"
              x1="0"
              x2="0"
              y1="66.625"
              y2="66.625"
            />
            <line
              class="bb-ygrid"
              x1="0"
              x2="0"
              y1="43.1875"
              y2="43.1875"
            />
            <line
              class="bb-ygrid"
              x1="0"
              x2="0"
              y1="19.75"
              y2="19.75"
            />
          </g>
          <g
            class="bb-xgrid-focus"
          >
            <line
              class="bb-xgrid-focus"
              style="visibility: hidden;"
              x1="-10"
              x2="-10"
              y1="0"
              y2="226"
            />
          </g>
        </g>
        <g
          class="bb-chart"
          clip-path="url(http://localhost/#bb-1560811908868-clip)"
        >
          <g
            class="bb-event-rects bb-event-rects-single"
            style="fill-opacity: 0;"
          >
            <rect
              class=" bb-event-rect bb-event-rect-0"
              height="226"
              width="0"
              x="0"
              y="0"
            />
            <rect
              class=" bb-event-rect bb-event-rect-1"
              height="226"
              width="0"
              x="0"
              y="0"
            />
            <rect
              class=" bb-event-rect bb-event-rect-2"
              height="226"
              width="0"
              x="0"
              y="0"
            />
          </g>
          <g
            class="bb-chart-bars"
          >
            <g
              class="bb-chart-bar bb-target bb-target-data1  "
              style="opacity: 0; pointer-events: none;"
            >
              <g
                class=" bb-shapes bb-shapes-data1 bb-bars bb-bars-data1"
                style="cursor: pointer;"
              />
            </g>
            <g
              class="bb-chart-bar bb-target bb-target-data2  "
              style="opacity: 0; pointer-events: none;"
            >
              <g
                class=" bb-shapes bb-shapes-data2 bb-bars bb-bars-data2"
                style="cursor: pointer;"
              />
            </g>
          </g>
          <g
            class="bb-chart-lines"
          >
            <g
              class="bb-chart-line bb-target bb-target-data1  "
              style="opacity: 0; pointer-events: none;"
            >
              <g
                class=" bb-shapes bb-shapes-data1 bb-lines bb-lines-data1"
              >
                <path
                  class=" bb-shape bb-shape bb-line bb-line-data1 bb-line-dashed-2-2"
                  d="M0,19.75L0,207.25L0,183.8125"
                  style="stroke: #4B9BFF; opacity: 1;"
                />
              </g>
              <g
                class=" bb-shapes bb-shapes-data1 bb-areas bb-areas-data1"
              >
                <path
                  class=" bb-shape bb-shape bb-area bb-area-data1"
                  style="fill: #4B9BFF;"
                />
              </g>
              <g
                class=" bb-shapes bb-shapes-data1 bb-circles bb-circles-data1"
                style="cursor: pointer;"
              >
                <circle
                  class=" bb-shape bb-shape-0 bb-circle bb-circle-0"
                  cx="0"
                  cy="19.75"
                  r="2.5"
                  style="fill: #4B9BFF; stroke: #4B9BFF; opacity: 1;"
                />
                <circle
                  class=" bb-shape bb-shape-1 bb-circle bb-circle-1"
                  cx="0"
                  cy="207.25"
                  r="2.5"
                  style="fill: #4B9BFF; stroke: #4B9BFF; opacity: 1;"
                />
                <circle
                  class=" bb-shape bb-shape-2 bb-circle bb-circle-2"
                  cx="0"
                  cy="183.8125"
                  r="2.5"
                  style="fill: #4B9BFF; stroke: #4B9BFF; opacity: 1;"
                />
              </g>
            </g>
            <g
              class="bb-chart-line bb-target bb-target-data2  "
              style="opacity: 0; pointer-events: none;"
            >
              <g
                class=" bb-shapes bb-shapes-data2 bb-lines bb-lines-data2"
              >
                <path
                  class=" bb-shape bb-shape bb-line bb-line-data2 bb-line-dashed-2-3"
                  d="M0,207.25L0,90.0625L0,19.75"
                  style="stroke: #FFB46E; opacity: 1;"
                />
              </g>
              <g
                class=" bb-shapes bb-shapes-data2 bb-areas bb-areas-data2"
              >
                <path
                  class=" bb-shape bb-shape bb-area bb-area-data2"
                  style="fill: #FFB46E;"
                />
              </g>
              <g
                class=" bb-shapes bb-shapes-data2 bb-circles bb-circles-data2"
                style="cursor: pointer;"
              >
                <rect
                  class=" bb-shape bb-shape-0 bb-circle bb-circle-0"
                  height="5"
                  style="fill: #FFB46E; stroke: #FFB46E; opacity: 1;"
                  width="5"
                  x="-2.5"
                  y="204.75"
                />
                <rect
                  class=" bb-shape bb-shape-1 bb-circle bb-circle-1"
                  height="5"
                  style="fill: #FFB46E; stroke: #FFB46E; opacity: 1;"
                  width="5"
                  x="-2.5"
                  y="87.5625"
                />
                <rect
                  class=" bb-shape bb-shape-2 bb-circle bb-circle-2"
                  height="5"
                  style="fill: #FFB46E; stroke: #FFB46E; opacity: 1;"
                  width="5"
                  x="-2.5"
                  y="17.25"
                />
              </g>
            </g>
          </g>
          <g
            class="bb-chart-arcs"
            transform="translate(0, 108)"
          />
        </g>
        <g
          class="bb-axis bb-axis-x"
          clip-path="url(http://localhost/#bb-1560811908868-clip-xaxis)"
          style="visibility: visible; opacity: 1;"
          transform="translate(0, 226)"
        >
          <text
            class="bb-axis-x-label"
            dx="-0.5em"
            dy="-0.5em"
            style="text-anchor: end;"
            x="0"
          />
          <g
            class="tick"
            style="opacity: 1;"
            transform="translate(0, 0)"
          >
            <line
              x1="0"
              x2="0"
              y2="0"
            />
            <text
              style="text-anchor: middle; display: block;"
              x="0"
              y="9"
            >
              <tspan
                dx="0"
                dy=".71em"
                x="0"
              >
                0
              </tspan>
            </text>
          </g>
          <g
            class="tick"
            style="opacity: 1;"
            transform="translate(0, 0)"
          >
            <line
              x1="0"
              x2="0"
              y2="0"
            />
            <text
              style="text-anchor: middle; display: block;"
              x="0"
              y="9"
            >
              <tspan
                dx="0"
                dy=".71em"
                x="0"
              >
                1
              </tspan>
            </text>
          </g>
          <g
            class="tick"
            style="opacity: 1;"
            transform="translate(0, 0)"
          >
            <line
              x1="0"
              x2="0"
              y2="0"
            />
            <text
              style="text-anchor: middle; display: block;"
              x="0"
              y="9"
            >
              <tspan
                dx="0"
                dy=".71em"
                x="0"
              >
                2
              </tspan>
            </text>
          </g>
          <path
            class="domain"
            d="M0,6V0H0V6"
          />
        </g>
        <g
          class="bb-axis bb-axis-y"
          style="visibility: visible; opacity: 1;"
          transform="translate(0, 0)"
        >
          <text
            class="bb-axis-y-label"
            dx="-0.5em"
            dy="1.2em"
            style="text-anchor: end;"
            transform="rotate(-90)"
            x="0"
          />
          <g
            class="tick"
            style="opacity: 1;"
            transform="translate(0,208)"
          >
            <line
              x2="-6"
              y1="0"
              y2="0"
            />
            <text
              style="text-anchor: end;"
              x="-9"
              y="0"
            >
              0
              <tspan
                dx="0"
                dy="3"
                x="-9"
              >
                20
              </tspan>
            </text>
          </g>
          <g
            class="tick"
            style="opacity: 1;"
            transform="translate(0,184)"
          >
            <line
              x2="-6"
              y1="0"
              y2="0"
            />
            <text
              style="text-anchor: end;"
              x="-9"
              y="0"
            >
              0
              <tspan
                dx="0"
                dy="3"
                x="-9"
              >
                30
              </tspan>
            </text>
          </g>
          <g
            class="tick"
            style="opacity: 1;"
            transform="translate(0,161)"
          >
            <line
              x2="-6"
              y1="0"
              y2="0"
            />
            <text
              style="text-anchor: end;"
              x="-9"
              y="0"
            >
              0
              <tspan
                dx="0"
                dy="3"
                x="-9"
              >
                40
              </tspan>
            </text>
          </g>
          <g
            class="tick"
            style="opacity: 1;"
            transform="translate(0,137)"
          >
            <line
              x2="-6"
              y1="0"
              y2="0"
            />
            <text
              style="text-anchor: end;"
              x="-9"
              y="0"
            >
              0
              <tspan
                dx="0"
                dy="3"
                x="-9"
              >
                50
              </tspan>
            </text>
          </g>
          <g
            class="tick"
            style="opacity: 1;"
            transform="translate(0,114)"
          >
            <line
              x2="-6"
              y1="0"
              y2="0"
            />
            <text
              style="text-anchor: end;"
              x="-9"
              y="0"
            >
              0
              <tspan
                dx="0"
                dy="3"
                x="-9"
              >
                60
              </tspan>
            </text>
          </g>
          <g
            class="tick"
            style="opacity: 1;"
            transform="translate(0,91)"
          >
            <line
              x2="-6"
              y1="0"
              y2="0"
            />
            <text
              style="text-anchor: end;"
              x="-9"
              y="0"
            >
              0
              <tspan
                dx="0"
                dy="3"
                x="-9"
              >
                70
              </tspan>
            </text>
          </g>
          <g
            class="tick"
            style="opacity: 1;"
            transform="translate(0,67)"
          >
            <line
              x2="-6"
              y1="0"
              y2="0"
            />
            <text
              style="text-anchor: end;"
              x="-9"
              y="0"
            >
              0
              <tspan
                dx="0"
                dy="3"
                x="-9"
              >
                80
              </tspan>
            </text>
          </g>
          <g
            class="tick"
            style="opacity: 1;"
            transform="translate(0,44)"
          >
            <line
              x2="-6"
              y1="0"
              y2="0"
            />
            <text
              style="text-anchor: end;"
              x="-9"
              y="0"
            >
              0
              <tspan
                dx="0"
                dy="3"
                x="-9"
              >
                90
              </tspan>
            </text>
          </g>
          <g
            class="tick"
            style="opacity: 1;"
            transform="translate(0,20)"
          >
            <line
              x2="-6"
              y1="0"
              y2="0"
            />
            <text
              style="text-anchor: end;"
              x="-9"
              y="0"
            >
              0
              <tspan
                dx="0"
                dy="3"
                x="-9"
              >
                100
              </tspan>
            </text>
          </g>
          <path
            class="domain"
            d="M-6,1H0V226H-6"
          />
        </g>
      </g>
      <g
        transform="translate(0, 260)"
      >
        <g
          class="bb-legend-item bb-legend-item-data1"
          style="visibility: visible; cursor: pointer;"
        >
          <text
            style="pointer-events: none;"
            x="1.5"
            y="13"
          >
            data1
          </text>
          <rect
            class="bb-legend-item-event"
            height="4"
            style="fill-opacity: 0;"
            width="25"
            x="-12.5"
            y="-1"
          />
          <line
            class="bb-legend-item-tile"
            stroke-width="10"
            style="stroke: #4B9BFF; pointer-events: none;"
            x1="-14.5"
            x2="-4.5"
            y1="8"
            y2="8"
          />
        </g>
        <g
          class="bb-legend-item bb-legend-item-data2"
          style="visibility: visible; cursor: pointer;"
        >
          <text
            style="pointer-events: none;"
            x="6.5"
            y="17"
          >
            data2
          </text>
          <rect
            class="bb-legend-item-event"
            height="4"
            style="fill-opacity: 0;"
            width="15"
            x="-7.5"
            y="3"
          />
          <line
            class="bb-legend-item-tile"
            stroke-width="10"
            style="stroke: #FFB46E; pointer-events: none;"
            x1="-9.5"
            x2="0.5"
            y1="12"
            y2="12"
          />
        </g>
      </g>
    </svg>
    <div
      class="bb-tooltip-container"
      style="position: absolute; pointer-events: none; display: none;"
    />
  </div>
</div>
`;
